bin/propolis-server/src/lib/vm/state_driver.rs

```diff
@@ -504,6 +508,26 @@ where
         }
     }
 
+    fn pause(&mut self, pause_kernel: PauseKernelVmm) {
+        assert!(!self.paused);
+        probes::state_driver_pause!(|| ());
+        self.vcpu_tasks.pause_all();
+        self.controller.pause_entities();
+        if pause_kernel == PauseKernelVmm::Yes {
+            self.controller.pause_vm();
+        }
```

Comment:
Is there a reason not to issue the `VM_PAUSE`?  Sure it's somewhat meaningless for a halt (or similar), but skipping it here leaves room for trouble later if one makes a `pause(PauseKernelVmm:No)` call, followed by a `resume()` call later.

Reply:
Nah, this was just rote-equivalence translation from the old code to the new. Cleaned up in 53f1a2e.